### PR TITLE
Rotate ARImageTrackingActionsImpl base node, use dimensions for android addImage, center align android image, load trackables in bg

### PR DESCRIPTION
**Pull request by nickolanack**

### CODE REVIEW

**Comment by nickolanack** [comment]
There is a Quaternion to Euler rotation conversion included in this PR that is not related to the ImageTracking issue but I accidentally included it. - i need camera rotation for my own purposes... (It is useful although I would think google will just implement this at some point)

File: src/ar.android.ts [hunk 1]
```diff
@@ -338,10 +386,26 @@ export class AR extends ARBase {
...
+                            -90, 
+                            0,
+                            0
+                        )
+                    ));
+
+
+                    //TODO calculate actual proper scale factor. for ios, tracking images have a defined width - I 
+                    // believe the scale factor is measured-width/defined-width in ios...
+                    planeNode.setLocalScale(new (<any>com.google.ar.sceneform).math.Vector3(0.025, 0.025, 0.025));
```
Comment (nickolanack) [inline src/ar.android.ts]: I've added this scale factor for now, but it needs to be calculated as per the comment...
for the pokemon card Bulbasaur, the IOS defined width is 4 meters but it appears to be about 0.1m on my computer
so thats where 0.025 comes from

**Comment by nickolanack** [comment]
for addImage I added a Sizer - but it is width or height only... (not both)

**Comment by nickolanack** [comment]
with the scale factor (0.25) and sizer in addImage, the favorite icon should work (more or less) without platform specific sizing ie:

```js
 args.imageTrackingActions.addImage({
                image: `~/app/images/favorite-${pokemon.favorite ? 'on' : 'off'}.png`,
                dimensions: {
                    x: 3,
                    y: 3
                },
                position: {
                    x: 3.5 ,
                    y: -1.6,
                    z: 0
                },
                scale: 1,
                onTap: toggleFavorite
            }).then(node => favNode = node);
```

**Comment by nickolanack** [comment]
Updated to load tracking images in background and also call session.configure() after adding tracking image. tracked images should work on first attempt (not require opening AR twice...) but still has a long delay before camera renders

File: src/ar.android.ts (hunk 1 above)
Comment (nickolanack) [reply src/ar.android.ts]: Updated this to calculate scale factor but using 4m as defined width of image

**Comment by nickolanack** [comment]
center aligned android addImage node (was anchored to bottom)

**Comment by nickolanack** [comment]
Implemented the android playVideo method in ARImageTrackingActions also center aligned video nodes (was bottom aligned )

**Comment by nickolanack** [comment]
@EddyVerbruggen - I'm gonna leave it there for tonight, I can't see the model floating over Eevee.jpg, but the favorite icon and the video is working ok on android for me

**Comment by nickolanack** [comment]
we need a way to read image width in meters for android,  maybe it should look for Contents.json in folder...

**Comment by EddyVerbruggen** [comment]
Re: "we need a way to read image width in meters for android, maybe it should look for Contents.json in folder..." that's a good point. I actually ignored those dimensions until now. I just corrected the image widths in demo-pokemon and now I'll have to adjust a few scale/dimension properties in the .ts code to compensate as they were specified as "4 meter" while actually they are "5 centimeters" 🤣

**Comment by EddyVerbruggen** [comment]
At first I had trouble making Android recognize images but then I read ARCore prefers real life images (the things you actually "film") of at least 15x15 cm.. which is way larger that the Pokemon cards I'm using. So I printed one on A4/Letter size and it works much better. 😃

Most of the model/image sizes have been fixed on iOS, tomorrow I'll take a closer look at the Android scales and UI of the Pokemon demo app.

**Comment by nickolanack** [comment]
Oh nice!
Keep in mind that this line https://github.com/EddyVerbruggen/nativescript-ar/blob/4970fc80f08b7558d820199d777643211b5cc87d/src/ar.android.ts#L420 has that hard coded value of 4m width. But needs to be read from config - but you might want to set it to 0.05 temporarily since you updated those model widths - that way your demo app shouldn’t need android specific scale

...also that scale factor does not affect the video, just the favorite icon and models

**Comment by EddyVerbruggen** [comment]
Since we now have the models of a few Pokemon for both platforms (I used a cmd line tool to convert `.dae` to `.glb`) I've now added a few to the SolarSystem demo (tested on iOS, will need scale-fixes on Android).

Someone should warn Elon! 🙀

IMG_3936

**Comment by nickolanack** [comment]
Lol

**Comment by nickolanack** [comment]
android studio has a way to package an image tracking database. I'll try to add support for that later... It'll probably speed up loading time alot.

**Comment by nickolanack** [comment]
Screenshot_20190912-143914
Dang, that is android with just IOS dimensions/scale (stripped all android specifics in augment-card.component.ts)  and set definedWidth=0.05

**Pull request merged**